packages/next/src/client/components/router-reducer/apply-flight-data.ts

```diff
@@ -11,21 +11,22 @@ export function applyFlightData(
   wasPrefetched: boolean = false
 ): boolean {
   // The one before last item is the router state tree patch
-  const [treePatch, subTreeData, head /* , cacheNodeSeedData */] =
-    flightDataPath.slice(-4)
+  const [treePatch, cacheNodeSeedData, head] = flightDataPath.slice(-3)
```

Comment:
While you're touching this, should extract the `slices` call to some sort of helper or something? It's always been sort of error-prone and this is a good opportunity to clean it up

Reply:
(same for the random array accesses)

Reply:
I intentionally left these alone because we’re about to completely change this data structure anyway.

Reply:
But yes I agree we should use accessor functions for internally unsound data structures in the future